index.js

```diff
@@ -63,6 +61,38 @@ module.exports.extra = function (pkg, deps, options) {
   return missing
 }
 
+function getDetective (name) {
+  try {
+    return name
+      ? (typeof name === 'string' ? require(name) : name)
+      : require('detective')
+  } catch (e) {}
+}
+
+function noopDetective () {
+  return []
+}
+
+function getExtensions (extensions, detective) {
+  var result = {
+    '.js': getDetective(detective),
+    '.node': noopDetective,
+    '.json': noopDetective
+  }
+
+  if (Array.isArray(extensions)) {
+    extensions.forEach(function (extension) {
+      result[extension] = getDetective(detective)
```

Comment:
No need to do a new `require()` every time here, just move the `getDetective(detective)` to outside of the loop. Also makes it more obvious that all extensions will actually be handed the same detective